.github/workflows/go-darwin.yml

```diff
@@ -2,7 +2,7 @@ name: Go-Darwin
 on:
   push:
     branches:
-      - "*"
+      - "master"
   pull_request:
```

Comment:
I prefer to get rid of pull_request altogether. Just use `branch: ["*"]` (or the priettied yaml equivalent). Branch is a superset of pull_request, and I like to have CI run on all my branch pushes so I can verify CI passes before raising the PR (sometimes).

Reply:
That will not cover the PR from fork model and not be very inviting to non anz-bank members who can't push branches to this repo.

I've also made these actions a required status check now.

Reply:
I actually even think it should be:
```yaml
on:
  push:
    branches: [ master ]
  pull_request:
    branches: [ master ]
```

Reply:
Ok then. fair enough. give me some time. I need to read up on exactly who these "on" stanzas work.

Reply:
I disagree with pull_request being master only. Sometimes I use another branch as a base if I have stacked PRs, and I would still want CI to run.

The docs for the pull_request event are incomplete. The synchronize type is not documented anywhere I could find. I assume it means when a PR is changed, but it would have been nice to see that documented.

Reply:
ok, will leave as is, but I think I prefer the shortened yaml...